lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -242,71 +195,208 @@ public void OnNext(IActiveContext activeContext)
+                return;
+            }
+            Exceptions.Throw(new Exception(string.Format("Data Loading Context with Id: {0} failed", value.Id)), Logger);
+        }
+
+        /// <summary>
+        /// Specifies what to do if a task fails.
+        /// We throw the exception and fail IMRU unless IMRU job is already done.
+        /// </summary>
+        /// <param name="value"></param>
+        public void OnNext(IFailedTask value)
         {
-            Logger.Log(Level.Error, "Cannot currently handle the Exception in OnError function");
-            throw new NotImplementedException("Cannot currently handle exception in OneError", error);
+            if (_areImruTasksCompleted)
+            {
+                Logger.Log(Level.Info,
+                    string.Format("Task with Id: {0} failed but IMRU task is completed. So ignoring.", value.Id));
+                return;
+            }
+            Exceptions.Throw(new Exception(string.Format("Task with Id: {0} failed", value.Id)), Logger);
         }
 
         /// <summary>
-        /// Specfies what to do when driver is done
-        /// In this case do nothing
+        /// Specified what to do once we receive a running task
         /// </summary>
+        /// <param name="value"></param>
+        public void OnNext(IRunningTask value)
+        {
+            Logger.Log(Level.Info, string.Format("Received Running Task message from task with Id: {0}", value.Id));
+        }
+
+        public void OnError(Exception error)
+        {
+        }
+
         public void OnCompleted()
         {
         }
 
+        /// <summary>
+        /// Generates map task configuration given the active context. 
+        /// Merge configurations of all the inputs to the MapTaskHost.
+        /// </summary>
+        /// <param name="activeContext">Active context to which task needs to be submitted</param>
+        /// <param name="taskId">Task Id</param>
+        /// <returns>Map task configuration</returns>
+        private IConfiguration GetMapTaskConfiguration(IActiveContext activeContext, string taskId)
+        {
+            IConfiguration mapSpecificConfig;
+
+            if (!_perMapperConfiguration.TryPop(out mapSpecificConfig))
+            {
+                Exceptions.Throw(
+                    new IMRUBugException(string.Format("No per map configuration exist for the active context {0}",
+                        activeContext.Id)),
+                    Logger);
+            }
+
+            return TangFactory.GetTang()
+                .NewConfigurationBuilder(TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, taskId)
+                    .Set(TaskConfiguration.Task, GenericType<MapTaskHost<TMapInput, TMapOutput>>.Class)
+                    .Build(),
+                    _configurationManager.MapFunctionConfiguration,
+                    mapSpecificConfig)
+                .BindNamedParameter(typeof(InvokeGC), _invokeGC.ToString())
```

Comment:
This is beyond the scope of this jira
On Apr 19, 2016 7:45 PM, "Julia Wang" <notifications@github.com> wrote:

> In lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs
> <https://github.com/apache/reef/pull/952#discussion_r60343817>:
>
> > +            if (!_perMapperConfiguration.TryPop(out mapSpecificConfig))
> > +            {
> > +                Exceptions.Throw(
> > +                    new IMRUBugException(string.Format("No per map configuration exist for the active context {0}",
> > +                        activeContext.Id)),
> > +                    Logger);
> > +            }
> > +
> > +            return TangFactory.GetTang()
> > +                .NewConfigurationBuilder(TaskConfiguration.ConfigurationModule
> > +                    .Set(TaskConfiguration.Identifier, taskId)
> > +                    .Set(TaskConfiguration.Task, GenericType<MapTaskHost<TMapInput, TMapOutput>>.Class)
> > +                    .Build(),
> > +                    _configurationManager.MapFunctionConfiguration,
> > +                    mapSpecificConfig)
> > +                .BindNamedParameter(typeof(InvokeGC), _invokeGC.ToString())
>
> use
> .BindNamedParameter(GenericType.Class, _invokeGC.ToString())
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/reef/pull/952/files/8c300c9c291b4f8cbeda476ff2e20470bd4270e9#r60343817>
>